tests/utils.py

```diff
@@ -167,40 +167,33 @@ def __init__(self, log, server='simple_server.py',
 
 
   def _start_server(self, timeout, extra_cmd_args, popen_cwd):
-    """Start the server subprocess. Uses a retry mechanism
-    if the bind fails."""
+    """
+    Start the server subprocess.
+    If the process hasn't started a Timeout exception will be raised.
+    """
 
     success = False
-    retries = 0
     elapsed = 0
     start = time.time()
-    while not success and elapsed < timeout:
-      retries += 1
-      self._start_process(extra_cmd_args,  popen_cwd)
-
-      # loop until bind succeeds, server exits or we timeout
-      while elapsed < timeout:
-        if not self.is_process_running():
-          break
-        elif self._is_port_found_in_log():
-          # If the port is in the log, then the bind was successful.
-          success = True
-          break
+
+    self._start_process(extra_cmd_args, popen_cwd)
+
+    # loop until bind succeeds, server exits or we timeout
+    while elapsed < timeout:
+      if not self.is_process_running():
+        break
+      elif self._set_port_if_in_logs():
+        # If the port is in the log, then the bind was successful.
+        success = True
+        break
 
         time.sleep(0.01)
         elapsed = time.time() - start
```

Comment:
wrong indent now -- should not be inside the conditional?

Reply:
Yes, a good point.
Forgot to fix it.